ArduCopter/AP_Arming.cpp

```diff
@@ -425,10 +425,10 @@ bool AP_Arming_Copter::gps_checks(bool display_failure)
     #endif
 
     // check if flight mode requires GPS
-    bool mode_requires_gps = copter.flightmode->requires_GPS();
+    bool mode_requires_gps = copter.flightmode->requires_GPS() || fence_requires_gps || (copter.simple_mode == Copter::SimpleMode::SUPERSIMPLE);
```

Comment:
Really bad variable name.

```suggestion
    const bool gps_required = copter.flightmode->requires_GPS() || fence_requires_gps || (copter.simple_mode == Copter::SimpleMode::SUPERSIMPLE);
```